src/platform/forms-system/src/js/validation.js

```diff
@@ -232,10 +232,11 @@ export function errorSchemaIsValid(errorSchema) {
   return _.values(_.omit('__errors', errorSchema)).every(errorSchemaIsValid);
 }
 
-export function isValidForm(form, pageListByChapters) {
-  const pageConfigs = _.flatten(_.values(pageListByChapters));
+export function isValidForm(form, pageList) {
+  const pageListMap = new Map();
+  pageList.forEach(page => pageListMap.set(page.pageKey, page));
   const validPages = Object.keys(form.pages).filter(pageKey =>
-    isActivePage(_.find({ pageKey }, pageConfigs), form.data),
+    isActivePage(_.find({ pageKey }, pageList), form.data),
```

Comment:
`pageListByChapters` was like:
```js
const pageListByChapters = {
  chapterName: {
    pageName: { ... }
  }
}
```
So when we did `const pageConfigs = _.flatten(_.values(pageListByChapters))`, it ended up with the same structure as `pageList`. Again, the `mapStateToProps` added in `appStateData` to the `pageList`, though, so using that one is preferred.